libcontainer/notify_linux_v2.go

```diff
@@ -0,0 +1,127 @@
+// +build linux
+
+package libcontainer
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+func getOOMCount(path string) (int, error) {
+	if _, err := os.Lstat(path); err != nil {
+		return 0, err
+	}
+
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, err
+	}
+
+	oomCount := 0
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		arr := strings.Split(line, " ")
+		if len(arr) == 2 && (arr[0] == "oom" || arr[0] == "oom_kill") {
+			count, err := strconv.Atoi(arr[1])
+			if err == nil && count > oomCount {
+				oomCount = count
+			}
+		}
+	}
+	return oomCount, nil
+}
+
+func getPopulated(path string) (int, error) {
+	if _, err := os.Lstat(path); err != nil {
+		return 0, err
+	}
+
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, err
+	}
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		arr := strings.Split(line, " ")
+		if len(arr) == 2 && arr[0] == "populated" {
+			return strconv.Atoi(arr[1])
+		}
+	}
+	return 0, nil
+}
+
+func registerMemoryEventV2(cgDir string, evName string, cgEvName string) (<-chan struct{}, error) {
+	eventControlPath := filepath.Join(cgDir, evName)
+	cgEvPath := filepath.Join(cgDir, cgEvName)
+	fd, err := unix.InotifyInit()
+	if err != nil {
+		return nil, err
```

Comment:
I'm afraid the error you get is very basic so you need to wrap it
```go
return errors.Wrap("unable to init inotify: %v", err)
```
or something like that